src/firefly/plans/util.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+
+__all__ = ["time_converter"]
+
+
+def time_converter(total_seconds):
+    """
+    Convert time (in seconds) to a tuple of hours, minutes, seconds
+
+    Parameters
+    ==========
+    total_seconds (float/int)
+
+    Returns
+    ==========
+    tuple: hours, minutes, seconds
+    """
+
+    # use np.round instead round to prevent errors in round(float("nan"))
+    hours = np.round(total_seconds // 3600)
+    minutes = np.round((total_seconds % 3600) // 60)
+    seconds = np.round(total_seconds % 60)
+    return hours, minutes, seconds
+
+
+def is_valid_value(value):
+    """
+    Check if the value is considered valid for inclusion in metadata.
+    Valid values are non-None, non-false, and if they are collections, they should have a positive length.
+
+    Parameters
+    ==========
+        value (any): The value to check.
+
+    Returns
+    ==========
+        bool: True if the value is valid, False otherwise.
+    """
+    # Check if the value is None
+    if value is None:
+        return False
+    # Check if the value is a collection with length
+    if isinstance(value, (str, list, tuple, dict)):
```

Comment:
The docstring says this works for any collection, but you only test for 4 specific ones. Maybe you want to test whether *value* is an instance of the Collection class? Or else update the doc-string.

https://docs.python.org/3/library/collections.abc.html#collections.abc.Collection

Reply:
updated doc-string; since not sure whether other kind of empty collection will also be removed from metadata